### PR TITLE
[Select] add optional prefix field for Select options

### PR DESCRIPTION
**Pull request by Ipkhchan**
### WHY are these changes introduced?
- useful feature addition. We have a use case for showing icons next to labels in pos-channel. Icons provide visual re-enforcement of selected values at a quick glance. 

Screen Shot 2020-10-02 at 3 57 49 PM

### WHAT is this pull request doing?
- adds an optional prefix field on `StrictOption`. When specified, the React node passed in is displayed to the left of the option label.

<details>
<summary>Example of Select with icons</summary>

</details>

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React, {useState} from 'react';
import {CircleDisableMinor} from '@shopify/polaris-icons';

import {Page, Select, Icon, TextStyle} from '../src';

export function Playground() {
  const [selected, setSelected] = useState('approved');
  const options = [
    {
      label: 'yes',
      value: 'yes',
      prefix: <TextStyle variation="strong">any react element</TextStyle>,
    },
    {label: 'no', value: 'no', prefix: <Icon source={CircleDisableMinor} />},
    {label: 'maybe', value: 'maybe'},
  ];

  return (
    <Page title="Playground">
      <Select
        label="Is doggo"
        options={options}
        onChange={(val) => setSelected(val)}
        value={selected}
      />
    </Page>
  );
}

```

</details>

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

I’ve tophatted these changes in the following browsers:

- [x] Chrome latest
- [x] FF latest
- [x] Safari latest
- [x] Edge
- [x] In at least one of the above browsers, test both retina and non-retina displays
- [ ] iPhone (5/SE/X) (10+) Safari Mobile
- [ ] iPad (10+) Safari Mobile
- [x] Android device (5.x) Chrome

### CODE REVIEW

**Comment by BPScott** [comment]
In https://github.com/Shopify/polaris-react/pull/3313 @tleunen made the case that having a generic "prefix" prop is more more flexible than a specific "icon" prop. What do you think to keeping our APIs consistent and using "prefix" here too?

**Review by alex-page — APPROVED** [review]

**Review by pnitsch — APPROVED** [review]

File: UNRELEASED.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Enhancements
 
+- Added `iconSource` field to `options` prop on `Select` ([#3373](https://github.com/Shopify/polaris-react/pull/3373))
```
Comment (BPScott) [inline UNRELEASED.md:11]:
```suggestion
- Added `prefix` field to `options` prop on `Select` ([#3373](https://github.com/Shopify/polaris-react/pull/3373))
```

File: src/components/Select/Select.tsx [hunk 2]
```diff
@@ -133,10 +135,15 @@ export function Select({
 
   const selectedOption = getSelectedOption(normalizedOptions, value);
 
+  const optionIconMarkup = selectedOption.prefix && (
```
Comment (BPScott) [inline src/components/Select/Select.tsx:138]: This isn't just for icons anymore. Rename to `prefixMarkup`?

**Review by BPScott — CHANGES_REQUESTED** [review]
Could you add an example of the usage to the Readme so we get to see what this looks like in storybook / on the styleguide?

**Review by alex-page — APPROVED** [review]

File: .storybook/polaris-readme-loader.js [hunk 3]
```diff
@@ -222,6 +222,7 @@ import {
   ConversationMinor,
   DeleteMinor,
   CircleDisableMinor,
+  CircleTickOutlineMinor,
```
Comment (BPScott) [inline .storybook/polaris-readme-loader.js:225]: This is going to require updating an extra icon to the list of imports in polaris-styleguide's stuff too. I'd like to keep avoid adding to these lists as it is a pain to remove things later. Any chance you could use another icon that's already on the blessed list?

**Review by BPScott — CHANGES_REQUESTED** [review]

File: .storybook/polaris-readme-loader.js (hunk 3 above)
Comment (alex-page) [reply .storybook/polaris-readme-loader.js:225]: That seems like an issue with the styleguide. I'll fix it up though. Thanks Ben!

**Pull request merged**